### PR TITLE
{chem}[GCCcore/9.3.0] ATAT v3.36

### PR DESCRIPTION
**Pull request by rsdmse**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by rsdmse** [comment]
The prefix should've been
{chem}[GCCcore/system]

File: easybuild/easyconfigs/a/atat/atat-3.36-GCCcore-9.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Ruoshi Sun <rs7wz@virginia.edu>
+# Research Computing, University of Virginia
+# 2020-05-23
+
+easyblock = 'MakeCp'
+
+name = 'atat'
```
Comment (boegel) [inline easybuild/easyconfigs/a/atat/atat-3.36-GCCcore-9.3.0.eb:7]: @rsdmse We usually try to stick to the official software name, which would be `ATAT` in this case, do you mind changing that?

Note that this also implies changing the easyconfig file name to `ATAT-3.36-GCCcore-9.3.0.eb`, and the subdirectory it's located in to `a/ATAT`.

You can use `--update-pr` for this, where you can [prefix the name of an easyconfig file with `:` to indicate it should be deleted](https://easybuild.readthedocs.io/en/latest/Integration_with_GitHub.html#deleting-easyconfig-files-or-patches).

So, if `ATAT.eb` is the local fixed easyconfig that uses `'ATAT'` as software name, this should work:

```
eb --update-pr 10674 :atat-3.36-GCCcore-9.3.0.eb ATAT.eb
```

You can use `eb --dry-run --update-pr 10674 ...` to verify this before taking any action.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, x86_64, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/22c4bfdc7de2b2b732d329f03b152291 for a full test report.

File: easybuild/easyconfigs/a/atat/atat-3.36-GCCcore-9.3.0.eb (hunk 1 above)
Comment (rsdmse) [reply easybuild/easyconfigs/a/atat/atat-3.36-GCCcore-9.3.0.eb]: @boegel Thanks for the tips. I changed the name to ATAT. The `BINDIR` variable is only invoked during `make install` but I see that a nonexistent `BINDIR` is causing the failure in the `make` stage in your test (even though the installation was successful locally and on Travis). I had to export it separately in `prebuildopts` and `postinstallcmds`, which isn't very clean but I couldn't find another way.

Before this commit I tried to do the customized make install commands in `install_cmd` but that was ignored by EasyBuild. (Although `eb -a -e MakeCp` shows `install_cmd` is a valid option in MakeCp.) So I had to put those under `postinstallcmds`.

Comment (rsdmse) [reply easybuild/easyconfigs/a/atat/atat-3.36-GCCcore-9.3.0.eb]: I forgot to change `%(name)s` to `%(namelower)s` but it still passed the test

**Comment by boegel** [comment]
> The `BINDIR` variable is only invoked during `make install` but I see that a nonexistent `BINDIR` is causing the failure in the `make` stage in your test (even though the installation was successful locally and on Travis). I had to export it separately in `prebuildopts` and `postinstallcmds`, which isn't very clean but I couldn't find another way.
> 
> Before this commit I tried to put the customized install commands in `install_cmd` but that was ignored by EasyBuild. (Although `eb -a -e MakeCp` shows `install_cmd` is a valid parameter in MakeCp.) So I had to put those under `postinstallcmds`.

The fact that `install_cmd` is still listed as a valid easyconfig parameter is indeed confusing, please open an issue in https://github.com/easybuilders/easybuild-easyblocks/issues for this, we should get that cleaned up.

I'm not sure `MakeCp` is the best choice of easyblock here, since you seem to suggest that there's a proper `make install` supported by `ATAT`?

Maybe using `ConfigureMake` together with `skipsteps = ['configure']` is a better fit, since then you can use `installopts` rather than `postinstallcmds`:

```python
preinstallopts = "export BINDIR=%(installdir)s/bin && "
installopts = "-C src && make install -C glue/jobctrl && make install -C glue/vasp"
```

The `make clean` is then no longer needed anymore, since we're not copying the whole build directory in the installation directory anymore (so `files_to_copy` is also useless).

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, x86_64, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/1e4b9ae7dbfa5bbc37ae007aa96bd7cb for a full test report.

**Comment by boegel** [comment]
@rsdmse The `BINDIR` issue is still there for me, it looks like `export BINDIR` isn't working, we need to override the `BINDIR` value by passing an argument to `make`, so something like:

```python
prebuildopts = "mkdir -p %(installdir)s/bin && "
buildopts = "BINDIR=%(installdir)s/bin"

**Comment by rsdmse** [comment]
Thanks again for your help. I discovered that I can't reference `%(installdir)s` in the build step, so I had to do `mkdir %(builddir)s/bin`, use that as BINDIR, and later copy `bin` to `%(installdir)s` as a `postinstallcmd`. It should work this time, hopefully!

**Comment by rsdmse** [comment]
I don't understand what's causing this error?
```
....E.E./opt/hostedtoolcache/Python/3.7.7/x64/lib/python3.7/site-packages/pep8.py:110: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
```

**Comment by rsdmse** [comment]
If possible please use the previous commit. (I deleted and re-forked the easyconfig repo, and now I can't update PR because of a mismatch in the branch name.)

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node379.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/3ba6422a02d517c554027f0d0be9f95c for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node051.manticore.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2650L v2 @ 1.70GHz, Python 2.7.5
See https://gist.github.com/76ba92a809e6ab56952b8f29b7083ffa for a full test report.

**Review by lexming — DISMISSED** [review]
LGTM

**Comment by lexming** [comment]
@boegel what do you think? Could you repeat your failed test?

**Comment by rsdmse** [comment]
I need to add an MIT license as a comment header. When I ran `eb --update-pr` everything went through successfully on the command line, but I can't see anything here.

**Comment by rsdmse** [comment]
@boegel The requested changes have been made and all checks have passed

**Pull request closed without merging**